.github/workflows/crate-io.yml

```diff
@@ -0,0 +1,32 @@
+---
+on:
+  release:
+    types: [created]
+    tags:
```

Comment:
We probably don't want to trigger on v-tags. Those are reserved for the CLN releases itself, and we are doing our best to make versions independent (multi-version schema migrations, etc)

Reply:
Ideally we'd want to compare the published version with the version in the Cargo.toml and trigger a run if the version was increased.

Reply:
I used the `ignore-unpublished-changes: true` so if the version is already published the CI is still happy.

Do you think that this does the job that you want?

Reply:
The problem is we're still compiling and testing the entire crate only to then try to upload and fail. Wasteful, but I guess quite alright :+1: We can always amend later.

Reply:
>The problem is we're still compiling and testing the entire crate only to then try to upload and fail.

Ah yes this is true